arch/sim/src/sim/posix/sim_hostfs.c

```diff
@@ -233,7 +246,7 @@ nuttx_ssize_t host_read(int fd, void *buf, nuttx_size_t count)
       ret = -errno;
     }
 
-  return ret;
+  return host_errno_convert(ret);
```

Comment:
this truncates nuttx_ssize_t to int. ditto for write, lseek.

Reply:
Thanks for your review, I changed it to convert errno and assign it to ret before returning it, so as to avoid truncates.